src/SFML/Window/WindowImpl.cpp

```diff
@@ -175,42 +176,59 @@ void WindowImpl::setMaximumSize(const std::optional<Vector2u>& maximumSize)
 
 
 ////////////////////////////////////////////////////////////
-Event WindowImpl::popEvent(bool block)
+Event WindowImpl::popEventImpl()
 {
+    Event event;
+
+    if (!m_events.empty())
+    {
+        event = m_events.front();
+        m_events.pop();
+    }
+
+    return event;
+}
+
+
+////////////////////////////////////////////////////////////
+Event WindowImpl::popEventBlocking(Time timeout)
+{
+    const auto startTime = std::chrono::steady_clock::now();
+
+    const auto timedOut = [&]
+    { return timeout != Time::Zero && (std::chrono::steady_clock::now() - startTime) < timeout.toDuration(); };
```

Comment:
Shouldn't that `<` become a `>=`? If `true` == `a timeout has occurred` then we want to return true when the actual duration exceeds or equals the timeout.

Reply:
Yes, this was a poor copy-paste mistake on my part while extracting the loop condition into a lambda.